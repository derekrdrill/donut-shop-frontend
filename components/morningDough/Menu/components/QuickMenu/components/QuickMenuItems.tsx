import * as React from 'react';
import { useRouter } from 'next/router';
import { Grid, Typography } from '@mui/material';

import { QuickMenuImage, QuickMenuItemContainer, QuickViewMenuItemsRoot } from '../style';

import { FullMenuItem } from '../../../assets/data/FULL_MENU';

interface QuickMenuItemsProps {
  quickMenuData: FullMenuItem[];
  setIsQuickMenuOpen: React.Dispatch<React.SetStateAction<boolean>>;
}

const QuickMenuItems = ({ quickMenuData, setIsQuickMenuOpen }: QuickMenuItemsProps) => {
  const router = useRouter()

  return (
    <QuickViewMenuItemsRoot container>
      {quickMenuData.map(menuItem => (
        <QuickMenuItemContainer
          key={menuItem.key}
          item
          onClick={
            /* istanbul ignore next */
            () => {
              setIsQuickMenuOpen(false);
              router.push(`/menu/${menuItem.subCategory}/${menuItem.key}`);
            }
          }
          xs={2}
        >
          <Grid container>
            <Grid item xs={12}>
              <Grid container justifyContent='center'>
                <QuickMenuImage src={menuItem.image} />
              </Grid>
            </Grid>
            <Grid item xs={12}>
              <Grid container justifyContent='center'>
                <Typography variant='subtitle1'>{menuItem.name}</Typography>
              </Grid>
            </Grid>
          </Grid>
        </QuickMenuItemContainer>
      ))}
    </QuickViewMenuItemsRoot>
  );
};

export default QuickMenuItems;
