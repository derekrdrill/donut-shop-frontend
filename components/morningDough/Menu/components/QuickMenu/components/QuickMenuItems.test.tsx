jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import QuickMenuItems from './QuickMenuItems';

const quickMenuData = [
  {
    key: 'coffee',
    name: 'Coffee',
    image: 'coffee.png',
    category: 'hotDrinks',
    subCategory: 'coffee',
  },
];

describe('QuickMenuItems unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(<QuickMenuItems quickMenuData={quickMenuData} setIsQuickMenuOpen={jest.fn()} />)
      .toJSON();
    expect(result).toMatchSnapshot();
  });
});
