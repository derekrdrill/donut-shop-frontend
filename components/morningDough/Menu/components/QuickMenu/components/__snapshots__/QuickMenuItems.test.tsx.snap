// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickMenuItems unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-container sc-hLseeU exfbdM css-11lq3yg-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 sc-dmqHEX hfdwKv css-1o7apob-MuiGrid-root"
    onClick={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
        >
          <img
            className="sc-beqWaB ETGuZ"
            src="coffee.png"
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          >
            Coffee
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;
